PSReadLine/Movement.cs

```diff
@@ -90,25 +102,53 @@ private void MoveToLine(int numericArg)
                         : point.X;
             }
 
-            var topLine = _initialY;
+            int newCurrent;
```

Comment:
Given all the new logic added and the single integer output, this might be worth putting into a new method.

Reply:
I feel the logic is too simple/small to put in a separate method ...